src/pages/index.js

```diff
@@ -6,87 +6,230 @@ import {
+      const newCardData = { _id, name, link, isLiked };
+      cardSection.addItem(createCard(newCardData));
+      addFormValidator.disableBtn();
+      cardAddForm.reset();
+      cardAddFormClass.close();
+    })
+    .catch((err) => {
+      console.error(`Error: failed to add card. ${err}`);
+    })
+    .finally(() => {
+      cardAddFormClass.isLoading(false);
+    });
 }
 
 function handleProfileFormSubmit(data) {
-  profileInfoClass.setUserInfo(data);
-  profileFormClass.close();
+  console.log(data);
+  profileFormClass.isLoading(true);
+  api
+    .setUserInfo(data)
+    .then(() => {
+      profileInfoClass.setUserInfo({ name: data.name, about: data.about });
+      editFormValidator.disableBtn();
+      profileFormClass.close();
+    })
+    .catch((err) => {
+      console.error(`Error handling profile form submit. ${err}`);
+    })
+    .finally(() => {
+      profileFormClass.isLoading(false);
+    });
 }
 
-function renderCard(card) {
-  const cardElementClass = new Card({
-    name: card.name,
-    link: card.link,
-    cardSelector: "#card-template",
-    handleImageClick: handleImageClick,
+function handlePictureSubmit(avatarUrl) {
+  pictureFormClass.isLoading(true);
+  api
+    .changeAvatar(avatarUrl)
+    .then(() => {
+      profileInfoClass.setUserInfo({ avatar: avatarUrl.picture });
+      pictureFormValidator.disableBtn();
+      pictureFormClass.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      pictureFormClass.isLoading(false);
+    });
+}
+
+function handleLikeBtn(card) {
+  api
+    .setCardLiked(card.id, card.getLikedStatus())
+    .then((updatedCard) => {
+      card.handleIsLiked(updatedCard.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleDeleteBtn(card) {
+  confirmFormClass.open();
+  confirmFormClass.handleConfirm(() => {
+    confirmFormClass.viewLoading(true);
+    api
+      .removeCard(card.id)
+      .then(() => {
+        card.removeCard();
+        confirmFormClass.viewLoading(false);
+        confirmFormClass.close();
+      })
+      .catch((err) => {
+        console.error(`Failed to delete card. ${err}`);
+        confirmFormClass.viewLoading(false);
+      });
   });
-  cardSectionClass.addItem(cardElementClass.getView());
 }
 
 // LISTENERS //
+pictureEditBtn.addEventListener("click", () => {
+  pictureFormValidator.resetValidation();
+  pictureLinkInput.value = pictureEl.src;
+  pictureFormClass.open();
+});
+
 profileEditBtn.addEventListener("click", () => {
-  const { name, description } = profileInfoClass.getUserInfo();
+  editFormValidator.resetValidation();
+  const { name, about } = profileInfoClass.getUserInfo();
   profileNameInput.value = name;
-  profileDescriptionInput.value = description;
+  profileDescriptionInput.value = about;
   profileFormClass.open();
 });
 
 cardAddBtn.addEventListener("click", () => {
+  addFormValidator.resetValidation();
```

Comment:
Method `resetValidation` is for clearing possible validation errors, when you open the **profile** form, because you insert **valid** data into its inputs.  


Please, delete it. You don't need it here